### ISSUE TITLE
Error parsing root validators

### ISSUE THREAD

**Issue by goroderickgo** [mansenfranzen/autodoc_pydantic#55]
Hello all, 

I really appreciate this Sphinx extension, but recently I've been running into some issues getting my docs to build. Would appreciate any guidance on what may be going on—from the traceback, it seems like my installation is getting tripped up trying to parse root validators. I tested a few versions of `autodoc_pydantic`, and it seems to appear in ≥ v.1.3

Here's the full traceback that Sphinx provided:

```python
# Sphinx version: 4.2.0
# Python version: 3.9.7 (CPython)
# Docutils version: 0.17.1 release
# Jinja2 version: 3.0.1
# Last messages:
#   failed: <lambda>() missing 1 required positional argument: 'aliased'
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 14 source files that are out of date
#   updating environment:
#   [new config]
#   14 added, 0 changed, 0 removed
#   reading sources... [  7%] index
#   reading sources... [ 14%] modules
#   reading sources... [ 21%] nmt
#   reading sources... [ 28%] nmt.common
# Loaded extensions:
#   sphinx.ext.mathjax (4.2.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (1.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (4.2.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc (4.2.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.graphviz (4.2.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/graphviz.py
#   sphinx.ext.imgmath (4.2.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/imgmath.py
#   sphinxcontrib.napoleon (0.7) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/napoleon/__init__.py
#   sphinxcontrib.autodoc_pydantic (1.4.0) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/__init__.py
#   furo (2021.09.22) from /opt/miniconda3/envs/nmt/lib/python3.9/site-packages/furo/__init__.py
Traceback (most recent call last):
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/application.py", line 343, in build
    self.builder.build_update()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 307, in build
    updated_docnames = set(self.read())
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 414, in read
    self._read_serial(docnames)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 435, in _read_serial
    self.read_doc(docname)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 475, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/parsers.py", line 101, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/directive.py", line 162, in run
    documenter.generate(more_content=self.content)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 983, in generate
    self.document_members(all_members)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 860, in document_members
    documenter.generate(
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 1773, in generate
    return super().generate(more_content=more_content,
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 980, in generate
    self.add_content(more_content)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/autodoc.py", line 241, in add_content
    self.add_field_summary()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/autodoc.py", line 331, in add_field_summary
    wrapper = ModelWrapper(self.object)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/inspection.py", line 87, in __init__
    self.field_validator_names = self.get_field_validator_names()
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/inspection.py", line 110, in get_field_validator_names
    field_names["*"].extend(self.get_names_from_root_validators())
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/inspection.py", line 121, in get_names_from_root_validators
    pre_root = get_name(self.model.__pre_root_validators__)
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/inspection.py", line 119, in get_name
    return [validator[1].__name__ for validator in validators]
  File "/opt/miniconda3/envs/nmt/lib/python3.9/site-packages/sphinxcontrib/autodoc_pydantic/inspection.py", line 119, in <listcomp>
    return [validator[1].__name__ for validator in validators]
TypeError: 'function' object is not subscriptable

```

**Comment by mansenfranzen** [comment]
@goroderickgo thanks for reporting the bug and the detailed report! Can you provide a minimal example of your pydantic model that causes the error, please? That would be great otherwise I'm having trouble to reproduce it. Additionally, can you check what pydantic version you are using? It could be related to a change in how pydantic stores and manages root validators.

**Comment by mansenfranzen** [comment]
@all-contributors please add @goroderickgo  for tests

**Comment by mansenfranzen** [comment]
It got fixed with version 1.5.0. Feel free to reopen of
